Documentation/design-docs/host-error-codes.md

```diff
@@ -0,0 +1,110 @@
+This is returned by `hostfxr_initialize_for_runtime_config` if it's called for the second time in a given process. Comes from `corehost_initialize` in `hostpolicy`.
+
+
+### Failure error/exit codes
+
+* `InvalidArgFailure` (`0x80008081`) - One of the specified arguments for the operation is invalid. This is used in a lot of places in the hosting layer.
+
+* `CoreHostLibLoadFailure` (`0x80008082`) - There was a failure loading a dependent library. If any of the hosting components calls `LoadLibrary`/`dlopen` on a dependent library and the call fails, this error code is returned. The most common case for this failure is if the dependent library is missing some of its dependencies (for example the necessary CRT is missing on the machine), likely corrupt or incomplete install.  
+This error code is also returned from `corehost_resolve_component_dependencies` if it's called on a `hostpolicy` which has not been initialized via the hosting layer. This would typically happen if `coreclr` is loaded directly without hosting layer and then `AssemblyDependencyResolver` is used (which is unsupported scenario).
+
+* `CoreHostLibMissingFailure` (`0x80008083`) - One of the dependent libraries is missing. Typically when the `hostfxr`, `hostpolicy` or `coreclr` dynamic libraries are not present in the expected locations. Probably means corrupted or incomplete installation.
+
+* `CoreHostEntryPointFailure` (`0x80008084`) - One of the dependent libraries is missing a required entry point. For example if 3.0 `hostfxr` is used via the `initialize_for_runtime_config` and the framework resolution logic ends up loading 2.0 `hostpolicy` which doesn't support the new behaviors.
+
+* `CoreHostCurHostFindFailure` (`0x80008085`) - If the hosting component is trying to use the path to the current module (the hosting component itself) and from it deduce the location of the installation. Either the location of the current module could not be determined (some weird OS call failure) or the location is not in the right place relative to other expected components.  
+For example the `hostfxr` may look at its location and try to deduce the location of the `shared` folder with the framework from it. It assumes the typical install layout on disk. If this doesn't work, this error will be returned.
+
+* `CoreClrResolveFailure` (`0x80008087`) - If the `coreclr` could not be found. The hosting layer (`hostpolicy`) looks for `coreclr` library either in the app itself (for self-contained) of in the root framework (for framework-dependent). This search can be done purely by looking at disk or more commonly by looking into the respective `.deps.json`. If `coreclr` is missing in `.deps.json` or it's there but doesn't exist on disk, this error is returned.
```

Comment:
Because coreclr is not a filename....
```suggestion
* `CoreClrResolveFailure` (`0x80008087`) - If the `coreclr` library could not be found. The hosting layer (`hostpolicy`) looks for `coreclr` library either in the app itself (for self-contained) of in the root framework (for framework-dependent). This search can be done purely by looking at disk or more commonly by looking into the respective `.deps.json`. If the `coreclr` library is missing in `.deps.json` or it's there but doesn't exist on disk, this error is returned.
```